### ISSUE TITLE
Using --device_index does not utilize the designated gpu for alignment and diarization

### ISSUE THREAD

**Issue by JarrodWoodard** [m-bain/whisperX#547]
TLDR: I pass my command with '--device_index 2' for instance, but when the job gets to alignment and diarization, device 0 is the one that is always used. 

I have three gpus, to speed up the amount of files I can process per hour I set up a script to spin up workers and assign transcription jobs to one of the three gpus if they are not currently working a job. This works as expected with the first three jobs, but gpu 0 always gets the alignment and diarization parts of all jobs and eventually throws errors. 

( tesla P40's, linux, pyannote/speaker-diarization@2.1, (pyannote 3 takes wayyy too long).
